circleguard/visualizer.py

```diff
@@ -180,9 +180,10 @@ def next_frame(self):
             return
 
         current_time = self.clock.get_time()
-        # resets visualizer if at end
+        # if we're at the end of the track or are at the beginning of the track
+        # (and thus are reversing), dont update
         if current_time > self.playback_len or current_time < 0:
-            self.reset(end=True if self.clock.current_speed < 0 else False)
+            return
```

Comment:
I would use `self.pause()` instead since the clock and everything is still progressing at this state.
Atm if you let it run after reaching the end and reverse the direction, you will have to wait till the clock enters the time range again.

Reply:
oh good point, I noticed that behavior but didn't think much of it